### PR TITLE
WebUI: Adds Codeformer support

### PR DESCRIPTION
**Pull request by psychedelicious**
Adds Codeformer support:
- Finished generalizing "gfpgan" to "facetool", this was partially done in a #1078 
- Facetool strength applies to both gfpgan and codeformer, codeformer fidelity added
- Metadata generation, reading and UI Metadata Viewer updated
- Bundle includes changes from #976 
- Users will need to click Settings/Gear icon > Reset Web UI to get the new values for facetool to load correctly (we should use redux-persist's migration feature to automatically do this kinda thing in the future)

Because Codeformer depends on the torch device, and I only have an Apple M1 device to use, this needs testing from somebody w/ a CUDA machine to test, just as a sanity check.

### CODE REVIEW

File: frontend/src/app/constants.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const PARAMETERS: { [key: string]: string } = {
   sampler: 'Sampler',
   seed: 'Seed',
   img2imgStrength: 'img2img Strength',
-  gfpganStrength: 'GFPGAN Strength',
+  facetoolStrength: 'GFPGAN Strength',
```
Comment (hipsterusername) [inline frontend/src/app/constants.ts:46]: Should rename to Facetool Strength

File: backend/invoke_ai_web_server.py [hunk 2]
```diff
@@ -611,23 +621,41 @@ def image_done(image, seed, first_seed):
...
+                            strength=facetool_parameters['strength'],
+                            seed=seed,
+                        )
+                    elif facetool_parameters['type'] == 'codeformer':
+                        image = self.codeformer.process(
+                            image=image,
+                            strength=facetool_parameters['strength'],
+                            fidelity=facetool_parameters['codeformer_fidelity'],
+                            seed=seed,
+                            device='cpu' if str(self.generate.device) == 'mps' else self.generate.device,
```
Comment (hipsterusername) [inline backend/invoke_ai_web_server.py:648]: FMI - Why is this here for codeformer but not for GFPGAN?

File: frontend/src/app/constants.ts (hunk 1 above)
Comment (psychedelicious) [reply frontend/src/app/constants.ts]: Updated with fix for this and a few other related places where things were named incorrectly.

File: backend/invoke_ai_web_server.py (hunk 2 above)
Comment (psychedelicious) [reply backend/invoke_ai_web_server.py:648]: I suppose it's needed bc codeformer can run on CUDA but not MPS, while GFPGAN can run on any device (or always on CPU - unsure). Doesn't work if you don't provide the device though.

**Review by hipsterusername — APPROVED** [review]
lgtm!

**Comment by discordinated** [comment]
First of all thanks for the work building this in, I love CodeFormer and run it locally on a lot of SD creations. 

While this PR is working for me, for some reason it's producing some really ugly output compared to CodeFormer running on my Mac locally. Perhaps I'm doing something wrong, 

Here's my initial test image:


Here is the output from Invoke/CodeFormer with Strength 1 and Fidelity 0
_(if I understand this right and please correct me if I don't - 'strength' is the % Invoke blends the CF output with the original image, so 1 should be 100% CF output and 0% original - Fidelity is the w slider in CF,  w=0 should be the same as Fidelity 0 and should be best quality output but less coherent to the original source image)_
The poor girl has some serious jaundice going on, overall the face now looks compressed / upsampled and is smooshy
001999 660639083 postprocessed

Here is the output from Invoke/CodeFormer with Strength 1 and Fidelity 1 for reference. 
Jaundice is gone but now it's very compressed / upsampled and smooshed looking
002002 660639083 postprocessed

Here's my local CodeFormer output using a selection of w=0 values and all 4 face detection models. I can't seem to replicate what is happening with the InvokeAI CF instance at all. Perhaps the Invoke version of CF is using an old model, or doesn't include all the new additions specific to high res faces in CF now? I checked codeformer.pth and it's the same version as my local install.

First the 2 high res face detection models: retinaface_resnet50 and YOLOv5l at w 0, upscale= 1
001818.660639083-w0-retinaface_resnet50-faceupsample
001818 660639083-w0-retinaface_resnet50-faceupsample

001818.660639083-w0-YOLOv5l-faceupsample
001818 660639083-w0-YOLOv5l-faceupsample

Then the 2 low res face detection models, again at w0 and upscale=1 
001818.660639083-w0-retinaface_mobile0.25_resnet-faceupsample
001818 660639083-w0-retinaface_mobile0 25_resnet-faceupsample

001818.660639083-w0-YOLOv5n-faceupsample
001818 660639083-w0-YOLOv5n-faceupsample

You may find it easier to download each image, or open in a new tab and cycle through them to see the differences. But all are visibly better than the InvokeAI CF output. Overall the results from the Invoke CF instance are not correct when compared to CF ran locally.  

Another slightly smaller face example, with a much more 'damaged' face:
Original


Invoke CF Str 1, Fid 0 -  This does look better at first then you notice she's now badly cross-eyed. 
002001 2305962182 postprocessed

And now 2 versions using resnet50 and YOLOv5l - both are a huge improvement over the InvokeAI implementation and neither has dramatically crossed her eyes. 

001954.2305962182-w0-retinaface_resnet50-faceupsample
001954 2305962182-w0-retinaface_resnet50-faceupsample

001954.2305962182-w0-YOLOv5l-faceupsample
001954 2305962182-w0-YOLOv5l-faceupsample

**Comment by hipsterusername** [comment]
@discordinated - Thank YOU for the deep research and sharing your testing. Would you mind creating a new issue? I think this is a deeper bug in the codeformer implementation, unrelated to the WebUI. 

Tag @lstein as well

**Comment by discordinated** [comment]
> @discordinated - Thank YOU for the deep research and sharing your testing. Would you mind creating a new issue? I think this is a deeper bug in the codeformer implementation, unrelated to the WebUI.
> 
> Tag @lstein as well

Posted https://github.com/invoke-ai/InvokeAI/issues/1167
